### DISCUSSION TITLE
Another but different sign in problem <unsolved>

### DISCUSSION THREAD

**Discussion by wantowan2** [community/community#70504] [New to GitHub]
I somehow lost my ability to push anything to github. I spent hours trying to figure it out. So i started over. 2 step, ssh keys, gpg keys, access tokens, authentication app. Now when i try to push something I get password box pop up. Ive tried my email pass, my token #, my computer logon, everything. Nothing works.  This is what i end up with:
`sign_and_send_pubkey: signing failed for ED25519 "wantowan2@gmail.com" from agent: agent refused operation
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
` Because I end up canceling the pop up

**Comment by queenofcorgis** [comment]
Have you ran `ssh-add` at all to add the SSH key to the agent? 

`ssh-add -l` is what will confirm if you have

**Reply by wantowan2** [reply]
ssh-add -l
The agent has no identities.
EDIT: So i went back a step to add which it did but its the same result. A popup box that i have no password for when i try to push a repo

**Comment by wantowan2** [comment]
This is where Im at"
OpenSSH_8.9p1 Ubuntu-3ubuntu0.4, OpenSSL 3.0.2 15 Mar 2022
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files
debug1: /etc/ssh/ssh_config line 21: Applying options for *
debug1: Connecting to github.com [140.82.112.3] port 22.
debug1: Connection established.
debug1: identity file /home/wantowan2/.ssh/id_rsa type -1
debug1: identity file /home/wantowan2/.ssh/id_rsa-cert type -1
debug1: identity file /home/wantowan2/.ssh/id_ecdsa type -1
debug1: identity file /home/wantowan2/.ssh/id_ecdsa-cert type -1
debug1: identity file /home/wantowan2/.ssh/id_ecdsa_sk type -1
debug1: identity file /home/wantowan2/.ssh/id_ecdsa_sk-cert type -1
debug1: identity file /home/wantowan2/.ssh/id_ed25519 type -1
debug1: identity file /home/wantowan2/.ssh/id_ed25519-cert type -1
debug1: identity file /home/wantowan2/.ssh/id_ed25519_sk type -1
debug1: identity file /home/wantowan2/.ssh/id_ed25519_sk-cert type -1
debug1: identity file /home/wantowan2/.ssh/id_xmss type -1
debug1: identity file /home/wantowan2/.ssh/id_xmss-cert type -1
debug1: identity file /home/wantowan2/.ssh/id_dsa type -1
debug1: identity file /home/wantowan2/.ssh/id_dsa-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_8.9p1 Ubuntu-3ubuntu0.4
debug1: Remote protocol version 2.0, remote software version babeld-dd067d10
debug1: compat_banner: no match: babeld-dd067d10
debug1: Authenticating to github.com:22 as 'git'
debug1: load_hostkeys: fopen /home/wantowan2/.ssh/known_hosts2: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256
debug1: kex: host key algorithm: ssh-ed25519
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: SSH2_MSG_KEX_ECDH_REPLY received
debug1: Server host key: ssh-ed25519 SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU
debug1: load_hostkeys: fopen /home/wantowan2/.ssh/known_hosts2: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
debug1: Host 'github.com' is known and matches the ED25519 host key.
debug1: Found key in /home/wantowan2/.ssh/known_hosts:1
debug1: rekey out after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey in after 134217728 blocks
debug1: get_agent_identities: bound agent to hostkey
debug1: get_agent_identities: ssh_fetch_identitylist: agent contains no identities
debug1: Will attempt key: /home/wantowan2/.ssh/id_rsa 
debug1: Will attempt key: /home/wantowan2/.ssh/id_ecdsa 
debug1: Will attempt key: /home/wantowan2/.ssh/id_ecdsa_sk 
debug1: Will attempt key: /home/wantowan2/.ssh/id_ed25519 
debug1: Will attempt key: /home/wantowan2/.ssh/id_ed25519_sk 
debug1: Will attempt key: /home/wantowan2/.ssh/id_xmss 
debug1: Will attempt key: /home/wantowan2/.ssh/id_dsa 
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp256-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,rsa-sha2-512-cert-v01@openssh.com,rsa-sha2-256-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,sk-ssh-ed25519@openssh.com,sk-ecdsa-sha2-nistp256@openssh.com,ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,rsa-sha2-512,rsa-sha2-256,ssh-rsa>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey
debug1: Next authentication method: publickey
debug1: Trying private key: /home/wantowan2/.ssh/id_rsa
debug1: Trying private key: /home/wantowan2/.ssh/id_ecdsa
debug1: Trying private key: /home/wantowan2/.ssh/id_ecdsa_sk
debug1: Trying private key: /home/wantowan2/.ssh/id_ed25519
debug1: Trying private key: /home/wantowan2/.ssh/id_ed25519_sk
debug1: Trying private key: /home/wantowan2/.ssh/id_xmss
debug1: Trying private key: /home/wantowan2/.ssh/id_dsa
debug1: No more authentication methods to try.
git@github.com: Permission denied (publickey).

**Comment by wantowan2** [comment]
- [ ] Its getting better
- [ ] `ssh -vT git@github.com
- [ ] OpenSSH_8.9p1 Ubuntu-3ubuntu0.4, OpenSSL 3.0.2 15 Mar 2022
- [ ] debug1: Reading configuration data /etc/ssh/ssh_config
- [ ] debug1: /etc/ssh/ssh_config line 19: include /etc/ssh/ssh_config.d/*.conf matched no files
- [ ] debug1: /etc/ssh/ssh_config line 21: Applying options for *
- [ ] debug1: Connecting to github.com [140.82.113.4] port 22.
- [ ] debug1: Connection established.
- [ ] debug1: identity file /home/wantowan2/.ssh/id_rsa type 3
- [ ] debug1: identity file /home/wantowan2/.ssh/id_rsa-cert type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_ecdsa type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_ecdsa-cert type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_ecdsa_sk type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_ecdsa_sk-cert type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_ed25519 type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_ed25519-cert type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_ed25519_sk type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_ed25519_sk-cert type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_xmss type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_xmss-cert type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_dsa type -1
- [ ] debug1: identity file /home/wantowan2/.ssh/id_dsa-cert type -1
- [ ] debug1: Local version string SSH-2.0-OpenSSH_8.9p1 Ubuntu-3ubuntu0.4
- [ ] debug1: Remote protocol version 2.0, remote software version babeld-70f1bac9
- [ ] debug1: compat_banner: no match: babeld-70f1bac9
- [ ] debug1: Authenticating to github.com:22 as 'git'
- [ ] debug1: load_hostkeys: fopen /home/wantowan2/.ssh/known_hosts2: No such file or directory
- [ ] debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
- [ ] debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
- [ ] debug1: SSH2_MSG_KEXINIT sent
- [ ] debug1: SSH2_MSG_KEXINIT received
- [ ] debug1: kex: algorithm: curve25519-sha256
- [ ] debug1: kex: host key algorithm: ssh-ed25519
- [ ] debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
- [ ] debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
- [ ] debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
- [ ] debug1: SSH2_MSG_KEX_ECDH_REPLY received
- [ ] debug1: Server host key: ssh-ed25519 SHA256:+DiY3wvvV6TuJJhbpZisF/zLDA0zPMSvHdkr4UvCOqU
- [ ] debug1: load_hostkeys: fopen /home/wantowan2/.ssh/known_hosts2: No such file or directory
- [ ] debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts: No such file or directory
- [ ] debug1: load_hostkeys: fopen /etc/ssh/ssh_known_hosts2: No such file or directory
- [ ] debug1: Host 'github.com' is known and matches the ED25519 host key.
- [ ] debug1: Found key in /home/wantowan2/.ssh/known_hosts:1
- [ ] debug1: rekey out after 134217728 blocks
- [ ] debug1: SSH2_MSG_NEWKEYS sent
- [ ] debug1: expecting SSH2_MSG_NEWKEYS
- [ ] debug1: SSH2_MSG_NEWKEYS received
- [ ] debug1: rekey in after 134217728 blocks
- [ ] debug1: get_agent_identities: bound agent to hostkey
- [ ] debug1: get_agent_identities: agent returned 1 keys
- [ ] debug1: Will attempt key: /home/wantowan2/.ssh/id_rsa ED25519 SHA256:JqToOquB9+yeCKBviK0lz2mh1XYWBT4Uh5D4MXqS6Hw agent
- [ ] debug1: Will attempt key: /home/wantowan2/.ssh/id_ecdsa 
- [ ] debug1: Will attempt key: /home/wantowan2/.ssh/id_ecdsa_sk 
- [ ] debug1: Will attempt key: /home/wantowan2/.ssh/id_ed25519 
- [ ] debug1: Will attempt key: /home/wantowan2/.ssh/id_ed25519_sk 
- [ ] debug1: Will attempt key: /home/wantowan2/.ssh/id_xmss 
- [ ] debug1: Will attempt key: /home/wantowan2/.ssh/id_dsa 
- [ ] debug1: SSH2_MSG_EXT_INFO received
- [ ] debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp256-cert-v01@openssh.com,sk-ssh-ed25519-cert-v01@openssh.com,sk-ecdsa-sha2-nistp256-cert-v01@openssh.com,rsa-sha2-512-cert-v01@openssh.com,rsa-sha2-256-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,sk-ssh-ed25519@openssh.com,sk-ecdsa-sha2-nistp256@openssh.com,ssh-ed25519,ecdsa-sha2-nistp521,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,rsa-sha2-512,rsa-sha2-256,ssh-rsa>
- [ ] debug1: SSH2_MSG_SERVICE_ACCEPT received
- [ ] debug1: Authentications that can continue: publickey
- [ ] debug1: Next authentication method: publickey
- [ ] debug1: Offering public key: /home/wantowan2/.ssh/id_rsa ED25519 SHA256:JqToOquB9+yeCKBviK0lz2mh1XYWBT4Uh5D4MXqS6Hw agent
- [ ] debug1: Server accepts key: /home/wantowan2/.ssh/id_rsa ED25519 SHA256:JqToOquB9+yeCKBviK0lz2mh1XYWBT4Uh5D4MXqS6Hw agent
- [ ] Authenticated to github.com ([140.82.113.4]:22) using "publickey".
- [ ] debug1: channel 0: new [client-session]
- [ ] debug1: Entering interactive session.
- [ ] debug1: pledge: filesystem
- [ ] debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
- [ ] debug1: client_input_hostkeys: searching /home/wantowan2/.ssh/known_hosts for github.com / (none)
- [ ] debug1: client_input_hostkeys: searching /home/wantowan2/.ssh/known_hosts2 for github.com / (none)
- [ ] debug1: client_input_hostkeys: hostkeys file /home/wantowan2/.ssh/known_hosts2 does not exist
- [ ] debug1: client_input_hostkeys: no new or deprecated keys from server
- [ ] debug1: Sending environment.
- [ ] debug1: channel 0: setting env LANG = "en_CA.UTF-8"
- [ ] debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
- [ ] Hi wantowan2! You've successfully authenticated, but GitHub does not provide shell access.
- [ ] debug1: channel 0: free: client-session, nchannels 1
- [ ] Transferred: sent 2156, received 2576 bytes, in 0.2 seconds
- [ ] Bytes per second: sent 13537.7, received 16174.9
- [ ] debug1: Exit status 1

`

**Comment by wantowan2** [comment]
but im still getting this:
`git push -u origin main
sign_and_send_pubkey: signing failed for ED25519 "/home/wantowan2/.ssh/id_rsa" from agent: agent refused operation
sign_and_send_pubkey: signing failed for ED25519 "wantowan2@gmail.com" from agent: agent refused operation
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
`

**Reply by Isuru-Eranda** [reply]
I have a Question, How you add a lable to your Question.

**Reply by wantowan2** [reply]
On your thread page there should be an edit button on the right side

**Reply by Isuru-Eranda** [reply]
But it is not there 😕 .Why has it happened ?

**Comment by Isuru-Eranda** [comment]
But it is not there 😕 .Why has it happened ?

**Comment by UZZAL321** [comment]
On Oct 23, 2023 6:52 AM, Isuru Eranda ***@***.***> wrote:
But it is not there 😕 .Why has it happened ?

**Comment by UZZAL321** [comment]
On Oct 23, 2023 8:28 AM, UZZAL KHAN ***@***.***> wrote:On Oct 23, 2023 6:52 AM, Isuru Eranda ***@***.***> wrote:
But it is not there 😕 .Why has it happened ?

**Comment by VanshGehlot** [comment]
I'd recommend for troubleshooting your GitHub push authentication issue:

1. Generate a new SSH key and add the public key to your GitHub account.
2. Make sure the SSH key is loaded in ssh-agent by running ssh-add ~/.ssh/id_rsa.
3. Verify the remote URL uses SSH format git@github.com: instead of HTTPS.
3. Check the SSH key fingerprint loaded matches your GitHub keys.
4. If still not working, try authenticating with HTTPS instead of SSH:  `git remote set-url origin https://github.com/user/repo`

When pushing, use your GitHub username/password for HTTPS authentication.

The main things to check are - SSH key setup, SSH url used, and loading the correct SSH key. Let me know if trying those steps helps resolve the push authentication problem!

**Reply by wantowan2** [reply]
I did what said and for some reason it was working. It was the second time yesterday, Howecer today its back to the old probs.
` Date: Thu Oct 1 09:23:02 2020 +0200
 1 file changed, 25 insertions(+), 21 deletions(-)
error: cannot run smimesign: No such file or directory
error: gpg failed to sign the data
error: failed to write commit object
fatal: cherry-pick failed
`
I had tried smimesign yesterday but it turned into needing this whole compiler so i stopped half way.

This is my .gitconfig filr, 
[user]
	email = wantowan2@gmail.com
	name = wantowan2
	signingkey = 8BBCDDF8D811584F
[color]
	ui = auto
[init]
	defaultBranch = development
[commit]
    sshsign = true
	gpgsign = false
	smimesign = false
[credential]
	helper = cache --timeout 3600
	credentialStore = secretservice
[safe]
	directory = /home/wantowan2/SM-T500/andoid/kernel/gta4lwifi

#[gpg "x509"]
	#program = smimesign
#[gpg]
	#format = x509

**Reply by wantowan2** [reply]
BUT! When i do ssh-add ~/.ssh/rsa, it fixes it probably until reboot.

**Comment by UZZAL321** [comment]
________________________________

**Comment by UZZAL321** [comment]
________________________________

**Comment by wantowan2** [comment]
I seem to be able to function on git, for the most part. Except for this cherry-pick I'm in the middle of. says:
`git status
On branch main
Your branch is up to date with 'origin/main'.

Cherry-pick currently in progress.
  (run "git cherry-pick --continue" to continue)
  (use "git cherry-pick --skip" to skip this patch)
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

nothing to commit, working tree clean
wantowan2@wantowan2-GE70-2PE:~/SM-T500/andoid/kernel/gta4lwifi$ git cherry-pick --continue
error: cannot run smimesign: No such file or directory
error: gpg failed to sign the data
error: failed to write commit object
fatal: cherry-pick failed
`It was from a time where ui was trying smimesign and gpg to make it work. Now I cant get rid of them.
